File: UI/Property Grid/Type Converters/BranchListConverter.vb
Public Class BranchListConverter : Inherits PgListConverter(Of GitProject, String)
    Protected Overrides Function GetListItems(Instance As GitProject, pd As PropertyDescriptor) As IEnumerable(Of String)
        If Instance IsNot Nothing Then
            Dim go As New GitObject(Instance.localRepository)
            Return go.ListBranches()
        Else
            Return {}
        End If
    End Function
End Class

File: UI/Property Grid/Type Converters/BranchListConverter.cs
using Grooper;
using System.Collections.Generic;
using System.ComponentModel;

namespace GrooperGit
{
    /// <summary>
    /// Creates a dropdown list of branches from the connected Git Project.
    /// </summary>
    public class BranchListConverter : PgListConverter<GitProject, string>
    {
        /// <inheritdoc/>
        protected override IEnumerable<string> GetListItems(GitProject Instance, PropertyDescriptor pd)
        {
            return Instance == null ? new List<string>() : (IEnumerable<string>)Instance.Repository.Branch("list").Split('\n');
        }
    }
}
